src/main/java/com/dape/api/adapter/repository/BetRepository.java

```diff
@@ -1,9 +1,22 @@
 package com.dape.api.adapter.repository;
 
 import com.dape.api.domain.entity.Bet;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface BetRepository extends JpaRepository<Bet, Long> {
+
+    @Query(
```

Comment:
Essa query não atende o filtro opcional dos campos. Se buscarmos pelo status apenas e não informar data de criação ou update, ela vai retornar apenas bets com o status informado e com datas nulas ou iguais ao valor passado (também nulo). Dá uma espiada em  Querydsl pra criar predicados dinamicos: https://spring.io/blog/2011/04/26/advanced-spring-data-jpa-specifications-and-querydsl

Temos implementado algo do tipo em nossas apis, posso te mostrar também